modules/auxiliary/gather/enum_azuresubdomains.rb

```diff
@@ -0,0 +1,155 @@
+          ".queue.core.windows.net",
+          ".table.core.windows.net",
+          ".mail.protection.outlook.com",
+          ".sharepoint.com",
+          ".redis.cache.windows.net",
+          ".documents.azure.com",
+          ".database.windows.net",
+          ".vault.azure.net",
+          ".azureedge.net",
+          ".search.windows.net",
+          ".azure-api.net",
+          ".azurecr.io", ]
+
+      # Array of keywords to prepend and append
+      permutations = [ "root",
+          "web",
+          "api",
+          "azure",
+          "azure-logs",
+          "data",
+          "database",
+          "data-private",
+          "data-public",
+          "dev",
+          "development",
+          "demo",
+          "files",
+          "filestorage",
+          "internal",
+          "keys",
+          "logs",
+          "private",
+          "prod",
+          "production",
+          "public",
+          "service",
+          "services",
+          "splunk",
+          "sql",
+          "staging",
+          "storage",
+          "storageaccount",
+          "test",
+          "useast",
+          "useast2",
+          "centralus",
+          "northcentralus",
+          "westcentralus",
+          "westus",
+          "westus2", ]
+
+      # Create permutated array of keywords and target domain
+      if datastore['PERMUTATIONS']
+        permutated_domains = Array.new()
+        for keywords in permutations do
+            permutated_domains.append(domain + "-" + keywords)
+            permutated_domains.append(keywords + "-" + domain)
+        end
+
+        # Normal list of subdomains
+        target_domains = Array.new()
+        for tld in subdomains do
+          target_domains.append(domain + tld)
+        end
+
+        # Permutated list of subdomains
+        for tld in subdomains do
+            for domain in permutated_domains do
+                target_domains.append(domain + tld)
+            end
+        end
+
+        # Query DNS records of permutated and normal target subdomains
+        for domain in target_domains do
+          begin
+            if Resolv.getaddress domain
+              print_status("Discovered Target Domain: #{domain} \n")
+              dns_get_a(domain) if datastore['ENUM_A']
+              dns_get_cname(domain) if datastore['ENUM_CNAME']
+              dns_get_ns(domain) if datastore['ENUM_NS']
+              dns_get_mx(domain) if datastore['ENUM_MX']
+              dns_get_soa(domain) if datastore['ENUM_SOA']
+              dns_get_txt(domain) if datastore['ENUM_TXT']
+            end
+            rescue
+              next
+            end
+          end
+      else
+
+        # Query DNS records of normal target subdomains
+        target_domains = Array.new()
+        for tld in subdomains do
+            target_domains.append(domain + tld)
+        end
+        for domain in target_domains do
+          begin
+            if Resolv.getaddress domain
```

Comment:
Using Metasploit native DNS resolution is preferred, if possible.

This should be possible with the `Exploit::Remote::DNS` or `Msf::Exploit::Remote::DNS::Enumeration` mixins.